### PR TITLE
Mongo Support

### PR DESCRIPTION
**Pull request by timkral**
@thespags 

Add MongoDB support. Also integrated with `MapStore`. It can be activated with:

```
mapStore.system="mongoDB"

mapStore.mongoDB.host="[host]"
mapStore.mongoDB.port=[port] (defaults to: 27017)
OR
mapStore.mongoDB.url="mongodb://[url]"
```
You can also specify an optional database name:
```
mapStore.mongoDB.database=[database]
```
If no database name is specified, it defaults to the application name.

### CODE REVIEW

**Comment by thespags** [comment]
@john-brock

File: mapstore-mongodb/src/main/scala/net/spals/appbuilder/mapstore/mongodb/MongoDBMapStorePlugin.scala [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    tableKey: MapStoreTableKey
+  ): Boolean = {
+    // Collection creation in MongoDB is not idempotent so we need to check that the tableName
+    // doesn't already exist in the collection name list
+    mongoDatabase.listCollectionNames().into(new java.util.ArrayList[String]()).contains(tableName) match {
+      // Case: Collection already exists
+      case true => true
+      // Case: Collection does not already exist
+      case false => {
+        // Turn of auto-indexing as we'll create our own explicit index
```
Comment (john-brock) [inline mapstore-mongodb/src/main/scala/net/spals/appbuilder/mapstore/mongodb/MongoDBMapStorePlugin.scala]: nit: turn **off**

**Review by thespags — APPROVED** [review]
Build failed dude.

**Comment by thespags** [comment]
You say mapStore.mongoDB.database=[database] can be provided but can it be set via the mongo url such as
mapStore.mongoDB.url = "mongodb://localhost:27017/drunkr"

where drunkr is my db.

**Comment by timkral** [comment]
If you need that functionality, can you log a feature request?

**Comment by thespags** [comment]
My question is it supported or not. I’ll make an issue if so but I don’t want to make an issue if it is.

**Comment by timkral** [comment]
No, that's not supported.

**Pull request merged**